jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/BufferedResponseHandler.java

```diff
@@ -53,14 +52,12 @@
  * generated can also be unbounded.
  * </p>
  */
-public class BufferedResponseHandler extends Handler.Wrapper
+public class BufferedResponseHandler extends ConditionalHandler
```

Comment:
I don't like the direct extension of `ConditionalHandler`.
Should it not extend `SkipNext`, as this is a "wrapper" `ConditionalHandler`?

Reply:
No.    The primary usage of `ConditionalHandler` should be extension, so `BufferedResponseHandler` is-a `ConditionalHandler` and `QoSHandler` is-a `ConditionalHandler` etc.    They implement `doHandle(...)` and optional `doNotHandle` to define how they behave if conditions are met or not met.

We provide `ConditionalHandler.SkipNext` as a concrete implementation to be used in XML assembly to make a handler that is not-a `ConditionalHandler` into one.     Perhaps the `Forbidden` and `DontHandle` concrete implementations are noise? although I can also see them being used in some XML assembly to achieve the desired structure without writing code.

Reply:
See the renames and logic inversion done after @janbartel's review.  I think naming makes more sense now.